src/renderer/components/badge/badge.tsx

```diff
@@ -16,26 +16,70 @@ export interface BadgeProps extends React.HTMLAttributes<any>, TooltipDecoratorP
 @withTooltip
 @observer
 export class Badge extends React.Component<BadgeProps> {
-  @observable isExpanded = false;
+  @observable _isExpanded = false;
+  @observable hasHighlightedText = false;
+  @observable interval?: NodeJS.Timeout;
```

Comment:
Still overcomplicated such simple thing and I'm against using timers where they really not necessary.
Otherwise to detect element size changes we could use `css-element-queries` (already in package.json) or `ResizeObserver` api (supported in chrome, but there is no types yet in TS afaik).

https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
https://caniuse.com/?search=ResizeObserver

Reply:
I think you are misunderstanding what that interval is doing. During the period of time when the mouse has been clicked (and only then) this interval periodically checks too see if any text has been selected. This has nothing to do with resizing the object so I don't see how `ResizeObserver` relates.

Reply:
But it should do something in ideal world with `ResizeObserver` since it detects when element size changes (badges might be with dynamic width / `%`).
About periodically checking via interval I can't get why we need this as a function? Use-case for me was obscure 
Otherwise for dealing with `document.getSelection()` I would recommend to use `document.addListener("selectionchange")`

Reply:
Interesting point, I tried directly setting `document.onselectionchange` since that is what the MDN docs seemed to imply but I was getting weird inconsistencies between the functions not seeing the same values. I would indeed much prefer an "interrupt" based system versus polling.

Let me do some more investigation.

Reply:
And still, why it's not enough just to check a selected text at `onMouseUp` before doing something (show/hide)?

Reply:
Well two fold, once I implemented this, it did fix the close on "decided not to highlight any text" situation as well as it doesn't close it when the click would normally clear the current selection. Both of these situations, imo, make the badge act a **lot** more consistently with how other systems work.

Reply:
Done, thanks for the suggestion.